src/components/SearchBar.js

```diff
@@ -1,22 +1,19 @@
-import { useState } from 'react'
 import { AiOutlineSearch } from 'react-icons/ai'
 import CollegeResult from '../app/components/CollegeResult'
-import searchAlgorithm from '@/utils/searchAlgorithm'
+import searchAlgorithm from '../utils/searchAlgorithm'
+import { useState } from 'react'
+
 const SearchBar = ({ data }) => {
   const [open, setOpen] = useState(false)
   const [filterData, setFilterData] = useState([])
-
   const handleSearch = (e) => {
     let input = e.target.value.trim().toLowerCase()
-    const filterData = searchAlgorithm(input)
+    const filterData = searchAlgorithm(input, data).sort((a, b) => b.score - a.score)
```

Comment:
I'm a bit skeptical with the `sort()` here. Maybe we can use it inside the search algorithm, in case that's the case, let's communicate with @ayushtiwari110 regarding this, we might need to write a new test for it, that'll drive this functionality.